src/api/spec/models/project_spec.rb

```diff
@@ -408,20 +409,67 @@
   describe '#self.valid_name?' do
     context "invalid" do
       it{ expect(Project.valid_name?(10)).to be(false) }
-      it{ expect(Project.valid_name?('home:M0ses:raspi::qtdesktop')).to be(false) }
+      it "has ::" do
+        property_of {
+          s1 = sized(1){ string(/[a-zA-Z0-9]/) }
+          s2 = sized(range(1, 199)){ string(/[-+\w\.]/) }
+          s = s1 + s2
+          i = range(0, (s.length - 2))
```

Comment:
If I read correctly, s would end with ::. In that case you wouldn't need a s3 var.

Basically I am looking for a nicer way to write this. Without having to juggle with indices and lots of variables.

I am sure there are better ways to write this, but something like this is what I had in mind

```ruby
string = [
  sized(1){ string(/[a-zA-Z0-9]/) },
  sized(range(1, 197)){ string(/[-+\w\.]/) },
  '::'
].flatten
```